From 84a9aeb9dd8340b0c2958bf35d9ec5bd230fe6d6 Mon Sep 17 00:00:00 2001
From: Yong Ding <yongding@codeaurora.org>
Date: Thu, 29 Mar 2018 17:40:35 +0800
Subject: [PATCH] soc: qcom: hab: move API func declarations to uapi

This is the temp solution before user-space hab clients
switch to use habmm.h in mm-hab.

Change-Id: Ib2d01c870fcf14543e5535a13527b2d681e33b90
Signed-off-by: Yong Ding <yongding@codeaurora.org>
---
 include/linux/habmm.h      | 36 ------------------------------------
 include/uapi/linux/habmm.h | 31 +++++++++++++++++++++++++++++++
 2 files changed, 31 insertions(+), 36 deletions(-)

diff --git a/include/linux/habmm.h b/include/linux/habmm.h
index 1c27ee4..b2715e7 100644
--- a/include/linux/habmm.h
+++ b/include/linux/habmm.h
@@ -11,39 +11,3 @@
  *
  */
 #include <uapi/linux/habmm.h>
-
-#ifndef _HABMM_H
-#define _HABMM_H
-
-int32_t habmm_socket_open(int32_t *handle, uint32_t mm_ip_id,
-			  uint32_t timeout, uint32_t flags);
-int32_t habmm_socket_close(int32_t handle);
-int32_t habmm_socket_send(int32_t handle, void *src_buff, uint32_t size_bytes,
-			  uint32_t flags);
-int32_t habmm_socket_recv(int32_t handle, void *dst_buff, uint32_t *size_bytes,
-			  uint32_t timeout, uint32_t flags);
-int32_t habmm_socket_sendto(int32_t handle, void *src_buff, uint32_t size_bytes,
-			    int32_t remote_handle, uint32_t flags);
-int32_t habmm_socket_recvfrom(int32_t handle, void *dst_buff,
-			      uint32_t *size_bytes, uint32_t timeout,
-			      int32_t *remote_handle, uint32_t flags);
-int32_t habmm_export(int32_t handle, void *buff_to_share, uint32_t size_bytes,
-		     uint32_t *export_id, uint32_t flags);
-int32_t habmm_unexport(int32_t handle, uint32_t export_id, uint32_t flags);
-int32_t habmm_import(int32_t handle, void **buff_shared, uint32_t size_bytes,
-		     uint32_t export_id, uint32_t flags);
-int32_t habmm_unimport(int32_t handle, uint32_t export_id, void *buff_shared,
-		       uint32_t flags);
-
-struct hab_socket_info {
-	int32_t vmid_remote; /* habmm's vmid */
-	int32_t vmid_local;
-	/* name from hypervisor framework if available */
-	char    vmname_remote[12];
-	char    vmname_local[12];
-};
-
-int32_t habmm_socket_query(int32_t handle, struct hab_socket_info *info,
-		uint32_t flags);
-
-#endif
diff --git a/include/uapi/linux/habmm.h b/include/uapi/linux/habmm.h
index b1b6561f..4b62f5e 100644
--- a/include/uapi/linux/habmm.h
+++ b/include/uapi/linux/habmm.h
@@ -3,6 +3,37 @@
 
 #include <linux/types.h>
 
+int32_t habmm_socket_open(int32_t *handle, uint32_t mm_ip_id,
+			  uint32_t timeout, uint32_t flags);
+int32_t habmm_socket_close(int32_t handle);
+int32_t habmm_socket_send(int32_t handle, void *src_buff, uint32_t size_bytes,
+			  uint32_t flags);
+int32_t habmm_socket_recv(int32_t handle, void *dst_buff, uint32_t *size_bytes,
+			  uint32_t timeout, uint32_t flags);
+int32_t habmm_socket_sendto(int32_t handle, void *src_buff, uint32_t size_bytes,
+			    int32_t remote_handle, uint32_t flags);
+int32_t habmm_socket_recvfrom(int32_t handle, void *dst_buff,
+			      uint32_t *size_bytes, uint32_t timeout,
+			      int32_t *remote_handle, uint32_t flags);
+int32_t habmm_export(int32_t handle, void *buff_to_share, uint32_t size_bytes,
+		     uint32_t *export_id, uint32_t flags);
+int32_t habmm_unexport(int32_t handle, uint32_t export_id, uint32_t flags);
+int32_t habmm_import(int32_t handle, void **buff_shared, uint32_t size_bytes,
+		     uint32_t export_id, uint32_t flags);
+int32_t habmm_unimport(int32_t handle, uint32_t export_id, void *buff_shared,
+		       uint32_t flags);
+
+struct hab_socket_info {
+	int32_t vmid_remote; /* habmm's vmid */
+	int32_t vmid_local;
+	/* name from hypervisor framework if available */
+	char    vmname_remote[12];
+	char    vmname_local[12];
+};
+
+int32_t habmm_socket_query(int32_t handle, struct hab_socket_info *info,
+		uint32_t flags);
+
 struct hab_send {
 	__u64 data;
 	__s32 vcid;
-- 
1.9.1

